totalRP3/modules/NamePlates/NamePlates.lua

```diff
@@ -0,0 +1,1124 @@
+local CONFIG_NAMEPLATES_ENABLE_CUSTOMIZATIONS = "nameplates_enable_customizations";
+local CONFIG_NAMEPLATES_ONLY_IN_CHARACTER     = "nameplates_only_in_character";
+local CONFIG_NAMEPLATES_SHOW_PLAYER_NAMES     = "nameplates_show_player_names";
+local CONFIG_NAMEPLATES_SHOW_PET_NAMES        = "nameplates_show_pet_names";
+local CONFIG_NAMEPLATES_SHOW_COLORS           = "nameplates_show_colors";
+local CONFIG_NAMEPLATES_SHOW_OOC_INDICATORS   = "nameplates_show_ooc_indicators";
+local CONFIG_NAMEPLATES_SHOW_ICONS            = "nameplates_show_icons";
+local CONFIG_NAMEPLATES_SHOW_TITLES           = "nameplates_show_titles";
+local CONFIG_NAMEPLATES_OOC_INDICATOR         = "nameplates_ooc_indicator";
+local CONFIG_NAMEPLATES_ACTIVE_QUERY          = "nameplates_active_query";
+
+-- Returns true if the given unit token refers to a combat companion pet.
+--
+-- Returns false if the unit is invalid, refers to a player, or is a
+-- battle pet companion.
+local function UnitIsCombatPet(unitToken)
+	-- Ensure battle pets don't accidentally pass this test, in case one
+	-- day they get nameplates added for no reason.
+	if UnitIsBattlePetCompanion(unitToken) then
+		return false;
+	end
+
+	return UnitIsOtherPlayersPet(unitToken) or UnitIsUnit(unitToken, "pet");
+end
+
+-- Returns the TRP3 API internal "unit ID" for a given unit token. This will
+-- typically be a "name-realm" string for a player, or a "name-realm_pet"
+-- for a companion pet.
+local function GetRegisterUnitID(unitToken)
+	if UnitIsPlayer(unitToken) then
+		return TRP3_Utils.str.getUnitID(unitToken);
+	elseif UnitIsCombatPet(unitToken) then
+		local companionType = TRP3_UI.misc.TYPE_PET;
+		return TRP3_UI.misc.getCompanionFullID(unitToken, companionType);
+	end
+end
+
+-- Returns the player model associated with the given unit token.
+--
+-- If no valid model can be found, nil is returned.
+local function GetPlayerProfile(unitToken)
+	local name, realm = UnitName(unitToken)
+	if not name or name == "" or name == UNKNOWNOBJECT then
+		-- Don't return profiles for invalid/unknown units.
+		return nil;
+	end
+
+	return Player.CreateFromNameAndRealm(name, realm);
+end
+
+-- Returns the (combat) pet companion profile associated with the given
+-- unit token.
+--
+-- If no profile can be found, nil is returned.
+local function GetCombatPetProfile(unitToken)
+	local companionType = TRP3_UI.misc.TYPE_PET;
+	local fullID = TRP3_UI.misc.getCompanionFullID(unitToken, companionType);
+	if not fullID then
+		return nil;
+	end
+
+	local profile = TRP3_Companions.register.getCompanionProfile(fullID);
+	if not profile then
+		return nil;
+	end
+
+	return profile.data;
+end
+
+-- Returns an appropriate default OOC indicator style for configuration.
+local function GetDefaultOOCIndicator()
+	local defaultOOCIndicator = "TEXT";
+
+	-- The default OOC icon should inherit from the tooltip setting if
+	-- available, but without adding a module dependency please.
+	if TRP3_Configuration then
```

Comment:
It could be a good thing to move the OOC indicator setting outside of the tooltip settings so it can be shared across modules, like what was done with the increased color contrast option.
I would even make it [OCC] by default and move it to advanced settings. Only Lenwë uses the red dot.

Reply:
Moving tooltip settings as described is out of the scope of this feature. If another PR comes along to do that, I'll merge in the appropriate changes.